api/WebApi.ts

```diff
@@ -145,7 +145,10 @@ export class WebApi {
                 userAgent = `${nodeApiName} (${window.navigator.userAgent})`;
             }
         } else {
-            const nodeApiVersion: string = JSON.parse(fs.readFileSync(path.resolve(__dirname, 'package.json'), 'utf8')).version;
+            let nodeApiVersion: string = 'unknown';
+            if (fs.existsSync(path.resolve(__dirname, 'package.json'))) {
```

Comment:
any reason not to do
```
var packageFile = path.resolve(__dirname, 'package.json')
```
And use that in the exists and read calls?

Also, would it be better to not do the exists check and just do the read/parse inside a try/catch? There are other reasons you might not be able to read the package.json file. Do we still want to blow up on those?

Reply:
Updated to use a variable. What could cause us to fail to read the package.json file if it exists? All I can think of is that the file didn't get copied correctly, or permissions issues, either of which probably are going to cause other issues - I don't love silently not throwing on other stuff because I think the most likely thing that could cause is us missing something getting messed up on our end - otherwise I don't see any issues coming here.

Reply:
permission issues was mainly what I was thinking of. I'm fine blowing up in that case, because you are right, if it exists and we can't read it, it probably means things later down the line will blow up as well.